testing/wait_for_backend_and_run_e2etests.sh

```diff
@@ -6,4 +6,5 @@
 set -ex
 timeout 240 sh -c "while ! curl http://proxy:80/api/actuator/health/ping 2>/dev/null | grep -q UP; do echo 'backend server not yet there - retrying in 1s'; sleep 1; done; echo 'backend server responded'"
 timeout 240 sh -c "while ! curl http://proxy:80/api/actuator/health/skyminderDummyServer 2>/dev/null | grep -q UP; do echo 'skyminder dummy server not yet there - retrying in 1s'; sleep 1; done; echo 'skyminder dummy server responded'"
+timeout 240 sh -c "while ! curl http://proxy:80/api/actuator/health/edcDummyServer 2>/dev/null | grep -q UP; do echo 'edc dummy server not yet there - retrying in 1s'; sleep 1; done; echo 'edc dummy server responded'"
```

Comment:
Das finde ich so nicht gut 
* Wir wiederholen uns ziemlich - wir haben 3 mal `timeout 240 sh -c  <...> sleep 1; done; echo <...>`
* Wenn nach 239s das backend da ist, nach weiteren 239s der skyminderDummyServer, und nach weiteren 239s der edcDummyServer dann läuft alles durch, und wir haben drei mal fast 4 minuten gewartet. Eigentlich wollen wir doch 4 minuten warten, bis _alles_ da ist. 

Daher mein vorschlag: In diesem Script eine funktion schreiben die überprüft, ob alle Voraussetzungen erfüllt sind. Und diese wird dann in dem timeout verwendet.

Reply:
Stimmt. Aber was ich so wie es aktuell ist nett finde ist, dass du im Log auch siehst welcher Service genau nicht UP war, wenn es hier einen Error gibt.
Diese Information würde ja verloren gehen, wenn du es mit einem einzigen Check machst, oder? Was meint ihr?

Reply:
> 

Und wenn wir alle Voraussetzungen in einem Rutsch testen, sehe ich dann im Log, welche der einzelnen Voraussetzungen verantwortlich war (welcher Service DOWN war) ?

Reply:
> Das finde ich so nicht gut
> 
> * Wir wiederholen uns ziemlich - wir haben 3 mal `timeout 240 sh -c  <...> sleep 1; done; echo <...>`
> * Wenn nach 239s das backend da ist, nach weiteren 239s der skyminderDummyServer, und nach weiteren 239s der edcDummyServer dann läuft alles durch, und wir haben drei mal fast 4 minuten gewartet. Eigentlich wollen wir doch 4 minuten warten, bis _alles_ da ist.
> 
> Daher mein vorschlag: In diesem Script eine funktion schreiben die überprüft, ob alle Voraussetzungen erfüllt sind. Und diese wird dann in dem timeout verwendet.

Und wenn wir alle Voraussetzungen in einem Rutsch testen, sehe ich dann im Log, welche der einzelnen Voraussetzungen verantwortlich war (welcher Service DOWN war) ?

Reply:
>angefangen Shell Script von Philip zu evaluieren ob es klappt und es umzusetzen<

Reply:
fixed gemeinsam mit Siegfried im `wait_for_backend_and_run_e2etests.sh`

Bitte nochmal reviewen, da etwas abweichend von deinem Vorschlag umgesetzt.

Reply:
ja - gefällt mir! Passt so.